zosia_site/cloud_run.tf

```diff
@@ -0,0 +1,128 @@
+}
+
+resource "google_artifact_registry_repository" "zosia-repo" {
+  location      = local.region
+  repository_id = "zosia-repo"
+  description   = "Repository for zosia site production images"
+  format        = "DOCKER"
+
+  cleanup_policies {
+    id     = "keep-minimum-versions"
+    action = "KEEP"
+    most_recent_versions {
+      keep_count = 5
+    }
+  }
+}
+
+resource "google_cloud_run_v2_job" "migrate" {
+  name     = "migrate"
+  location = local.region
+
+  template {
+    template {
+      service_account = google_service_account.cloudrun_service_account.email
+
+      containers {
+        image   = local.docker_image_url
+        command = ["./scripts/migrate.sh"]
+
+        env {
+          name  = "GOOGLE_CLOUD_PROJECT"
+          value = local.project_id
+        }
+      }
+    }
+  }
+}
+
+resource "google_cloud_run_v2_job" "collectstatic" {
+  name     = "collectstatic"
+  location = local.region
+
+  template {
+    template {
+      service_account = google_service_account.cloudrun_service_account.email
+
+      containers {
+        image   = local.docker_image_url
+        command = ["./scripts/collectstatic.sh"]
+
+        env {
+          name  = "GOOGLE_CLOUD_PROJECT"
+          value = local.project_id
+        }
+
+        env {
+          name  = "GCS_BUCKET_NAME"
+          value = google_storage_bucket.static_files_bucket.name
+        }
+      }
+    }
+  }
+}
+
+resource "google_cloud_run_v2_service" "zosia_site" {
+  name     = "zosia"
+  location = local.region
+
+  template {
+    service_account = google_service_account.cloudrun_service_account.email
+
+    containers {
+      image   = local.docker_image_url
+      command = ["./scripts/start_prod_server.sh"]
+
+      env {
+        name  = "GOOGLE_CLOUD_PROJECT"
+        value = local.project_id
+      }
+
+      env {
+        name  = "GCS_BUCKET_NAME"
+        value = google_storage_bucket.static_files_bucket.name
+      }
+
+      # TODO: Add domain mapping to zosia.org and www.zosia.org
+      env {
+        name  = "HOSTS"
+        value = "zosia.org, www.zosia.org"
+      }
+    }
+  }
+}
+
+# This allows zosia website to be accessed by anyone without authentication
+resource "google_cloud_run_v2_service_iam_member" "noauth" {
+  location = google_cloud_run_v2_service.zosia_site.location
+  name     = google_cloud_run_v2_service.zosia_site.name
+  role     = "roles/run.invoker"
+  member   = "allUsers"
+}
+
+resource "random_id" "static_files_bucket_prefix" {
+  byte_length = 8
+}
+
+resource "google_storage_bucket" "static_files_bucket" {
+  name          = "${random_id.static_files_bucket_prefix.hex}-static-files-bucket"
+  location      = local.region
+  force_destroy = false
+  storage_class = "STANDARD"
+
+  cors {
+    origin          = ["*"]
+    method          = ["GET"]
+    response_header = ["*"]
+  }
```

Comment:
If this bucket is not set to act as a "website", then you don't need to define CORS.

Reply:
This bucket is public and is accessed directly by the end user's browser to download static files, so it won't work without defining CORS :/